dask/dataframe/groupby.py

```diff
@@ -1110,6 +1112,17 @@ def apply(self, func, *args, **kwargs):
         1.  The user should provide output metadata.
         2.  If the grouper does not align with the index then this causes a full
             shuffle.  The order of rows within each group may not be preserved.
+        3.  Dask's GroupBy.apply is not appropriate for functions reductions.
+            For custom reductions, use :class:`dask.dataframe.Aggregation`.
```

Comment:
From building the docs locally, it looks like sphinx wants the full path to `Aggregation` for linking to the class API documentation

```suggestion
            For custom reductions, use :class:`dask.dataframe.groupby.Aggregation`.
```